extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.js

```diff
@@ -101,56 +103,26 @@ angular.module('PaperUI.controllers', [ 'PaperUI.constants' ]).controller('BodyC
                         state = parsedValue;
                     }
                 }
-                if (item.type === "Rollershutter") {
-                    if (stateObject.type == "Percent" || stateObject.type == "Decimal") {
-                        state = parseInt(stateObject.value);
+                if (stateObject.type == "Percent" || stateObject.type == "Decimal") {
+                    if (item.type === "Rollershutter") {
+                        state = parseInt(state);
+                    } else {
+                        state = parseFloat(state);
                     }
                 }
+
+                updateState = updateState && item.state !== state;
                 if (updateState) {
-                    $scope.$apply(function(scope) {
-                        item.state = state;
-                        item.stateText = util.getItemStateText(item);
-                    });
+                    console.log('Updating ' + itemName + ' to ' + state)
+                    item.state = state;
+                    item.stateText = util.getItemStateText(item);
                 } else {
                     console.log('Ignoring state ' + state + ' for ' + itemName)
                 }
             }
-            if (item.members) {
-                $.each(item.members, function(i, memberItem) {
-                    changeStateRecursively(memberItem);
-                });
-            }
-        }
-        var index = getItemIndex(itemName);
-        if (index !== -1) {
-            changeStateRecursively($rootScope.data.items[index]);
         }
     });
 
-    eventService.onEvent('smarthome/items/*/statechanged', function(topic, stateObject) {
```

Comment:
I would have expected that you are actually adding a line to receive changed events and not removing it.
Aren't you know still listening for ALL state events and you do the detection of changes in the UI itself instead of merely listening on changed events?

Reply:
I refactored the `state` handling to now handle `statechanged`events. See https://github.com/eclipse/smarthome/pull/5092/files#diff-86bb251ac89e6ba6cda9d10aecc6664dR77

Reply:
Ah, thanks, I somehow overlooked that line above!